### PR TITLE
many: add new `snapd-observe` interface

### PR DESCRIPTION
**Pull request by mvo5**
A different approach (thanks Samuele!) for https://github.com/snapcore/snapd/pull/12675 

This provides a new `snapd-observe` interface that can be used from snaps to observe certain snap operations. Initially only `/v2/changes{,/**}`and `/v2/snaps{,/**}` is supported.

### CODE REVIEW

File: interfaces/builtin/snapd_observe.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+var supportedApis = []string{"/v2/changes"}
+
+func (iface *snapdObserveInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
+	api, ok := plug.Attrs["api"].(string)
```
Comment (pedronis) [inline interfaces/builtin/snapd_observe.go:48]: this would ideally be a list no?

**Review by pedronis — COMMENTED** [review]
I feel like we could start by giving access to `/v2/snaps{,<name>}` and `/v2/changes/*` and grow something like`api` possibly only later?

File: interfaces/builtin/snapd_observe.go (hunk 1 above)
Comment (mvo5) [reply interfaces/builtin/snapd_observe.go:48]: Yes, but I removed it now as suggested in the top-level comments

File: daemon/access.go [hunk 2]
```diff
@@ -160,7 +160,8 @@ var (
 	requireThemeApiAccess = requireThemeApiAccessImpl
 )
 
-func requireThemeApiAccessImpl(d *Daemon, ucred *ucrednet) *apiError {
+// TODO: not ideal api yet
+func requireSnapdSocketOrConnectedIface(d *Daemon, ucred *ucrednet, ifaceName, ifaceAttr, ifaceAttrVal string) *apiError {
```
[the remark this answers is not in the document]
Comment (mvo5) [reply daemon/access.go:164]: The signature is now simpler (just the ifaceName) so maybe less relevant now? But if you prefer I'm happy to still make this take a function of course.

**Comment by mvo5** [comment]
Closing for now as this probably needs to be redone when we get /v2/nocites

**Pull request closed without merging**